From 473af23f9d3c0cef55d4c4ed863247281946c614 Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Thu, 23 Jan 2025 12:37:13 +0530
Subject: [PATCH] QSDK:CP: wifi-script: Fix warning during new link add using
 wifi command

Configure wlan0 interface with 2 link mld as below

wlan0
2GHz
5GHz

Now add 6GHz link in wlan0 using wifi mlo_add_link command, below
warning print is seen
WARNING: Variable 'data' does not exist or is not an array/object

During new link addition we call drv_mac80211_setup() to create hostapd
config file for perticular link. In drv_mac80211_setup() we are getting
this warning with json_select data, this we don't need to run for new link
addition.

To fix this issue, avoid using json_select data when mlo link addition is
in progress.

Change-Id: Iec823cba17b85a818d7f2c8091f39f6dba17700f
Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
---
 .../files/lib/netifd/wireless/mac80211.sh           | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
index 26f66ea2f3b3..eb468b3749b3 100755
--- a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
+++ b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
@@ -1700,11 +1700,14 @@ drv_mac80211_setup() {
                 radio=-1
         fi
 
-	json_select data && {
-		json_get_var prev_rxantenna rxantenna
-		json_get_var prev_txantenna txantenna
-		json_select ..
-	}
+	[ -f /tmp/mlo_support.txt ] && mlo_add_flag=$(cat /tmp/mlo_support.txt)
+	if [ $mlo_add_flag -eq 0 ]; then
+		json_select data && {
+			json_get_var prev_rxantenna rxantenna
+			json_get_var prev_txantenna txantenna
+			json_select ..
+		}
+	fi
 
 	find_phy || {
 		echo "Could not find PHY for device '$1'"
-- 
2.34.1

